module/x/gravity/keeper/keeper.go

```diff
@@ -664,22 +664,20 @@ func (k Keeper) DeleteEthereumSignatures(ctx sdk.Context, otx types.OutgoingTx)
 // This just does keeper state cleanup if a new gravity contract has been deployed
 func (k Keeper) MigrateGravityContract(ctx sdk.Context, newBridgeAddress string, bridgeDeploymentHeight uint64) {
 	// Delete Any Outgoing TXs.
-
 	prefixStoreOtx := prefix.NewStore(ctx.KVStore(k.storeKey), []byte{types.OutgoingTxKey})
 	iterOtx := prefixStoreOtx.ReverseIterator(nil, nil)
 	defer iterOtx.Close()
 	for ; iterOtx.Valid(); iterOtx.Next() {
+		prefixStoreOtx.Delete(iterOtx.Key())
```

Comment:
I'm not sure if it support deletion while iteration.

Reply:
fixed